packages/core/src/runtime/Hook.ts

```diff
@@ -0,0 +1,31 @@
+export declare function afterAll(fnc: HookFn, timeout?: number): void
+export declare function afterEach(fnc: HookFn, timeout?: number): void
+export declare function beforeAll(fnc: HookFn, timeout?: number): void
+export declare function beforeEach(fnc: HookFn, timeout?: number): void
+
+export type HookFn = (this: void) => any
+export interface HookBase {
+	fnc: HookFn
+	timeout?: number
+}
+
+export interface Hook {
+	afterAll: HookBase[]
+	afterEach: HookBase[]
+	beforeAll: HookBase[]
+	beforeEach: HookBase[]
+}
+
+/**
+ * @internal
+ */
+export function normalizeHookBase(hookBase: HookBase): HookBase {
+	// Convert user inputted seconds to milliseconds
+	if (typeof hookBase.timeout === 'number' && hookBase.timeout > 1e3) {
+		hookBase.timeout = hookBase.timeout / 1e3
+	} else if (Number(hookBase.timeout) === 0 || !hookBase.timeout) {
+		hookBase.timeout = 30
```

Comment:
I think right now it should be
```suggestion
      const { timeout } = hookBase
	hookBase.timeout = timeout &&  timeout > 0 ? timeout : 30 * 1e3
```
in the next enhancement ticket, we can remove this function and the timeout value should evaluate by something likes
```
function captureHookWithAfterAll(...args: any[]) {
	const [fnc, timeout] = args
	hook.afterAll.push({ fnc, ms(timeout)})
}
```